tests/dynamic_data/base_test_util.py

```diff
@@ -106,7 +107,14 @@ def check_specific_parametrizations(rest_parametrization, specific_parametrizati
         assert len(sp_parametrization) == len(rest_parametrization), f"Specific parametrization must have same number of members as rest_paramertization, test_case={test_case_key}, attribute={attribute_name}, idx={idx}"
         for dimension in rest_parametrization:
             dimension_enum = get_dimension_enum(dimension)
-            param_values = [v for v in sp_parametrization if v in dimension_enum]
+            param_values = []
+            for parametrization_key in sp_parametrization:
+                if inspect.isclass(parametrization_key) and issubclass(parametrization_key, Enum):
+                    if parametrization_key == dimension_enum:
+                        param_values.append(parametrization_key)
+                else:
+                    if parametrization_key in dimension_enum:
+                        param_values.append(parametrization_key)
```

Comment:
if parametrization_key == dimension_enum:
                    param_values.append(parametrization_key)
                elif parametrization_key in dimension_enum:
                    param_values.append(parametrization_key)

param_values = [v for v in sp_parametrization if v == dimension_enum or v in dimension_enum]